pcsx2/USB/Win32/Config_usb.cpp

```diff
@@ -0,0 +1,244 @@
+ *  without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR
+ *  PURPOSE.  See the GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along with PCSX2.
+ *  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "PrecompiledHeader.h"
+#include "AppCoreThread.h"
+#include "../USB.h"
+#include "resource_usb.h"
+#include "Config_usb.h"
+#include "../deviceproxy.h"
+#include "../usb-pad/padproxy.h"
+#include "../usb-mic/audiodeviceproxy.h"
+#include "../configuration.h"
+#include "../shared/inifile_usb.h"
+
+HINSTANCE hInstUSB;
+extern bool configChanged;
+
+void SysMessageA(const char* fmt, ...)
+{
+	va_list list;
+	char tmp[512];
+
+	va_start(list, fmt);
+	vsprintf_s(tmp, 512, fmt, list);
+	va_end(list);
+	MessageBoxA(0, tmp, "USB Msg", 0);
+}
+
+void SysMessageW(const wchar_t* fmt, ...)
+{
+	va_list list;
+	wchar_t tmp[512];
+
+	va_start(list, fmt);
+	vswprintf_s(tmp, 512, fmt, list);
+	va_end(list);
+	MessageBoxW(0, tmp, L"USB Msg", 0);
+}
+
+void SelChangedAPI(HWND hW, int port)
+{
+	int sel = SendDlgItemMessage(hW, port ? IDC_COMBO_API1_USB : IDC_COMBO_API2_USB, CB_GETCURSEL, 0, 0);
+	int devtype = SendDlgItemMessage(hW, port ? IDC_COMBO1_USB : IDC_COMBO2_USB, CB_GETCURSEL, 0, 0);
+	if (devtype == 0)
+		return;
+	devtype--;
+	auto& rd = RegisterDevice::instance();
+	auto devName = rd.Name(devtype);
+	auto apis = rd.Device(devtype)->ListAPIs();
+	auto it = apis.begin();
+	std::advance(it, sel);
+	changedAPIs[std::make_pair(port, devName)] = *it;
+}
+
+void PopulateAPIs(HWND hW, int port)
+{
+	SendDlgItemMessage(hW, port ? IDC_COMBO_API1_USB : IDC_COMBO_API2_USB, CB_RESETCONTENT, 0, 0);
+	int devtype = SendDlgItemMessage(hW, port ? IDC_COMBO1_USB : IDC_COMBO2_USB, CB_GETCURSEL, 0, 0);
+	if (devtype == 0)
+		return;
+	devtype--;
+	auto& rd = RegisterDevice::instance();
+	auto dev = rd.Device(devtype);
+	auto devName = rd.Name(devtype);
+	auto apis = dev->ListAPIs();
+
+	std::string selApi = GetSelectedAPI(std::make_pair(port, devName));
+
+	std::string var;
+	if (!LoadSetting(nullptr, port, rd.Name(devtype), N_DEVICE_API, str_to_wstr(var)))
+	{
+		if (apis.begin() != apis.end())
+		{
+			selApi = *apis.begin();
+			changedAPIs[std::make_pair(port, devName)] = selApi;
+		}
+	}
+
+	int i = 0, sel = 0;
+	for (auto& api : apis)
+	{
+		auto name = dev->LongAPIName(api);
+		if (!name)
+			continue;
+		SendDlgItemMessageW(hW, port ? IDC_COMBO_API1_USB : IDC_COMBO_API2_USB, CB_ADDSTRING, 0, (LPARAM)name);
+		if (api == var)
+			sel = i;
+		i++;
+	}
+	SendDlgItemMessage(hW, port ? IDC_COMBO_API1_USB : IDC_COMBO_API2_USB, CB_SETCURSEL, sel, 0);
+}
+
+BOOL CALLBACK ConfigureDlgProcUSB(HWND hW, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+
+	int port;
+	switch (uMsg)
+	{
+		case WM_INITDIALOG:
+			SendDlgItemMessageA(hW, IDC_BUILD_DATE_USB, WM_SETTEXT, 0, (LPARAM)__DATE__ " " __TIME__);
```

Comment:
This additional build version is no longer needed: https://pastebin.com/KvkiUbvz